File: app/src/main/java/com/appspot/pistatium/ssmemo/models/Memo.java
package com.appspot.pistatium.ssmemo.models;

import java.io.Serializable;
import java.util.Date;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

/**
 * Created by kimihiro on 2015/11/08.
 */
public class Memo extends RealmObject implements Serializable {

    @PrimaryKey
    private Long id;
    private String title = "";
    private String memo = "";
    private Date createdAt = new Date();
    private Date updatedAt = new Date();
    private Date dueAt;
    private int priority = Priority.LOW.value;
    private int status = Status.ACTIVE.value;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getMemo() {
        return memo;
    }

    public void setMemo(String memo) {
        this.memo = memo;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    public Date getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Date updatedAt) {
        this.updatedAt = updatedAt;
    }

    public Date getDueAt() {
        return dueAt;
    }

    public void setDueAt(Date dueAt) {
        this.dueAt = dueAt;
    }

    public int getPriority() {
        return priority;
    }

    public void setPriority(int priority) {
        this.priority = priority;
    }

    public int getStatus() {
        return status;
    }

    public void setStatus(int status) {
        this.status = status;
    }
}

File: app/src/main/kotlin/com/appspot/pistatium/ssmemo/models/Memo.kt
package com.appspot.pistatium.ssmemo.models

import java.io.Serializable
import java.util.Date

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.RealmClass
import kotlin.properties.Delegates

/**
 * Created by kimihiro on 2015/11/08.
 */

@RealmClass
public open class Memo: RealmObject() {

    @PrimaryKey
    public open var id: Long = -1
    public open var title = ""
    public open var memo = ""
    public open var createdAt = Date()
    public open var updatedAt = Date()
    public open var dueAt: Date? = null
    public open var priority: Int = Priority.LOW.value
    public open var status: Int = Status.ACTIVE.value
}
